apps/api/src/middleware/authMiddleware.ts

```diff
@@ -55,21 +55,22 @@ type AuthenticationErrorResponse = ErrorResponse<['FORBIDDEN', 'UNAUTHORIZED']>;
  */
 export const authMiddleware =
 	(config: AuthMiddlewareConfig = {}): RequestHandler =>
-	(request, response: Response<AuthenticationErrorResponse>, next) => {
-		const { requiredRoles } = config;
+	async (request, response: Response<AuthenticationErrorResponse>, next) => {
 		const { user } = request.session;
+		const userRole = getUserRole(request.session);
+		const { requiredRoles } = config;
 
 		if (!user) {
 			response.status(401).send({
 				error: 'UNAUTHORIZED',
 				message: 'This resource is protected and requires authorization.',
 			});
+
 			return;
 		}
 
 		if (requiredRoles) {
-			const role = getUserRole(request.session);
-			if (!requiredRoles.includes(role)) {
+			if (!requiredRoles.includes(userRole)) {
```

Comment:
Looks like we can restore this entire function to how it originally was? We don't need to get the role until after we know there are required roles.